manifests/application.pp

```diff
@@ -50,7 +50,7 @@
   $enable_ostree = $katello::params::enable_ostree
   $enable_yum = $katello::params::enable_yum
   $enable_file = $katello::params::enable_file
-  $enable_puppet = $katello::params::enable_puppet
+  $enable_puppet = $katello::params::_enable_puppet
```

Comment:
Thinking more about it, should we fix it here:
```suggestion
  $enable_puppet = $katello::params::pulp2_support and $katello::params::enable_puppet
```

That avoids the additional variable.

Reply:
Sure, I really had no idea, this module becoming really hard to follow after the re-factor to params/globals/application. Hopefully we can just drop all of that organization looking ahead.

Reply:
The long term goal I had was to stop using init.pp altogether. Then `globals` would be exposed in kafo as well as `application` and `candlepin`. `params` wouldn't be. https://github.com/theforeman/foreman-installer/pull/421 would do that. I rebased it, but it needs migrations. However, what that allows us to do is to remove the `pulp` module from the answers file on EL8, which means you only have `--pulp-*` options on EL7. Due to different priorities I was never able to finish this work.